internal/customermanagedkeys/hsm_key_acceptance_template.go

```diff
@@ -0,0 +1,158 @@
+package customermanagedkeys
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-uuid"
+)
+
+// ManagedHSMKeyTempalte: Helper function to generate a template for HSM key acceptance tests
+// Ensure `azurerm_client_config.current` datasource is defined before using this template.
+// Verify there are no resource address conflicts in the caller of this template.
+func ManagedHSMKeyTempalte(randomInteger int, randomString string, purgeProtectionEnabled bool, principalRefs []string) string {
```

Comment:
Lets not do this, we should not be sharing test configuration between services as sometimes different services require slightly different setups. Not to mention it moves the test config away from the tests

so could we please change this to maybe `CosmosDBManagedHSMKeyTempalte` and include it in the cosmosdb service folder somewhere?